data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,617 @@
+- : int -> int = <fun>
+
+# fun s t -> s ^ " " ^ t ;;
+- : string -> string-> string = <fun>
+
+# function [] -> None | x :: _ -> Some x;;
+- : 'a list -> 'a option = <fun>
+```
+
+In order, here is what they do:
+- The identity function takes anything, returns it unchanged.
+- The square function takes an integer, returns its square.
+- The function taking two strings returns their concatenation with a space character in between.
+- A function taking a list returns `None`, if the list is empty, and the tail of the list otherwise.
+
+## Defining Global Functions
+
+Defining a global function isn't very different from creating a definition and binding a value to a name. The only difference is that no computation takes place. The expression is turned into a function value and bound to the name.
+```ocaml
+# let f = fun x -> x * x;;
+val f : int -> int = <fun>
+
+# let f x = x * x;;
+val f : int -> int = <fun>
+```
+
+These two definitions are the same. The former explicitly binds the anonymous function to a name. The latter is the same, but it uses a more compact syntax and avoids the `fun` keyword and the arrow symbol.
+
+## Defining Local Functions
+
+A function may be defined locally. Just like any local definition, meaning the function is only available inside its attached expression. Although the local functions are often defined inside the function's scope, this is not a requirement. Here is a local function inside a toplevel expression:
+```ocaml
+# let sq x = x * x in sq 7 * sq 7;;
+- : int = 2401
+
+# sq;;
+Error: Unbound value sq
+```
+
+## Closures
+
+*Closures* are directly related to shadowing. As introduced in [Same-Level Shadowing](#same-level-shadowing] section, a closure uses the preceding name binding, even after that name is redefined. In other words, it's *enclosed* between two different definitions; it will never compute the name binding that comes after it. That's why it is called a closure.
+
+The following sequence of definitions and expressions illustrates what a closure is:
+```ocaml
+# let d = 2 * 3;;
+val d : int = 6
+
+# let f x = x * d;;
+val f : int -> int = <fun>
+
+# f 7;;
+- : int = 42
+
+# let d = 10;;
+val d : int = 10
+
+# f 7;; (* What is the result? *)
+- : int = 42
+```
+
+Here is how this makes sense:
+1. Constant `d` is defined, and its value is 6.
+1. Function `f` is defined. It takes a single parameter and returns its product by `d`.
+1. Compute `f` of 7
+1. Create a new definition `d`, shadowing the first one
+1. Compute `f` of 7 again, the result is the same
+
+Although the new definition of `d` *shadows* the first one, its original value remains the one `f` uses. That first definition of `d` is trapped inside the definition of `f`. The environment is frozen as it was when `f` was defined, with the first value of `d`. A *function value* is a pair containing the function code and an environment. It is this pair that is called a [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)).
```

Comment:
“its original value remains” seems to imply an update took place. “the original remains” is more accurate.